hummingbot/market/kucoin/kucoin_api_order_book_data_source.py

```diff
@@ -0,0 +1,300 @@
+        """
+        async with aiohttp.ClientSession() as client:
+
+            market_response, exchange_response = await safe_gather(
+                client.get(TICKER_PRICE_CHANGE_URL),
+                client.get(EXCHANGE_INFO_URL)
+            )
+            market_response: aiohttp.ClientResponse = market_response
+            exchange_response: aiohttp.ClientResponse = exchange_response
+
+            if market_response.status != 200:
+                raise IOError(f"Error fetching Kucoin markets information. "
+                              f"HTTP status is {market_response.status}.")
+            if exchange_response.status != 200:
+                raise IOError(f"Error fetching Kucoin exchange information. "
+                              f"HTTP status is {exchange_response.status}.")
+
+            market_data = await market_response.json()
+            exchange_data = await exchange_response.json()
+            
+            attr_name_map = {"baseCurrency": "baseAsset", "quoteCurrency": "quoteAsset"}
+
+            trading_pairs: Dict[str, Any] = {item["symbol"]: {attr_name_map[k]: item[k] for k in ["baseAsset", "quoteAsset"]}
+                                             for item in exchange_data["data"]
+                                             if item["enableTrading"] == True}
+
+            market_data: List[Dict[str, Any]] = [{**item, **trading_pairs[item["symbol"]]}
+                                                 for item in market_data["data"]["ticker"]
+                                                 if item["symbol"] in trading_pairs]
+
+            # Build the data frame.
+            all_markets: pd.DataFrame = pd.DataFrame.from_records(data=market_data, index="symbol")
+            # calculating the correct USDVolume by multiplying or diving with necessary USD pair
+            # split_symbol = all_markets.split('-')
+            if all_markets.symbol.startswith("USDT"):
+                all_markets.loc[:, "USDVolume"] = 1 / all_markets.volvalue
+            elif all_markets.symbol.endswith("USDT"):
+                all_markets.loc[:, "USDVolume"] = all_markets.volvalue
+            else:
+                try:
+                    quote_currency: str = symbol.split('-')[1]
+                    mul: str = quote_currency + "-USDT"
+                    div: str = "USDT-" + quote_currency
+                    if all_markets.loc[mul]:
+                        all_markets.loc[:, "USDVolume"] = all_markets.volvalue * all_markets.loc[mul].last
+                    else:
+                        all_markets.loc[:, "USDVolume"] = all_markets.volvalue / all_markets.loc[div].last
+                except Exception:
+                    self.logger().error(f"Unable to get USD Volume for {trading_pair}. ", exc_info=True)
+            all_markets.loc[:, "volume"] = all_markets.quoteVolume
+            return all_markets.sort_values("USDVolume", ascending=False)
```

Comment:
I understand that this portion of code is meant to generate the `USDVolume` column for the `DataFrame` but there are definitely some stuff you need to fine tune. Considering that I am not getting any desirable output from the `get_active_exchange_markets()` function.

You might want to read up on the `pandas` library, in particular the documentation on [`DataFrame`](https://pandas.pydata.org/pandas-docs/stable/reference/frame.html).

Also you might want to refer to [Binance](https://github.com/CoinAlpha/hummingbot/blob/f8d1c19dfd0875bd12717f9c46ddbe20cc7b9a0d/hummingbot/market/binance/binance_api_order_book_data_source.py#L57), [Bittrex](https://github.com/CoinAlpha/hummingbot/blob/f8d1c19dfd0875bd12717f9c46ddbe20cc7b9a0d/hummingbot/market/bittrex/bittrex_api_order_book_data_source.py#L61) or [CoinbasePro](https://github.com/CoinAlpha/hummingbot/blob/f8d1c19dfd0875bd12717f9c46ddbe20cc7b9a0d/hummingbot/market/coinbase_pro/coinbase_pro_api_order_book_data_source.py#L56) to generate the relevant data.

Reply:
Solved. I've corrected the get_active_exchange_markets() to give correct USD Volume.